standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -470,28 +320,29 @@ private boolean checkIfTableIsUsable(String tableName, boolean configValue) thro
       // don't check it if disabled
       return false;
     }
-    Connection dbConn = null;
-    boolean tableExists = true;
+    jdbcResource.bindDataSource(POOL_TX);
     try {
-      dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED);
-      try (Statement stmt = dbConn.createStatement()) {
-        // Dummy query to see if table exists
-        try (ResultSet rs = stmt.executeQuery("SELECT 1 FROM \"" + tableName + "\"")) {
-          rs.next();
-        }
-      }
-    } catch (SQLException e) {
+      jdbcResource.getJdbcTemplate().query("SELECT 1 FROM \"" + tableName + "\"", new MapSqlParameterSource(),
+          rs -> {
+            if (rs.next()) {
```

Comment:
do we care about the result? i think if table doesn't exist, exception would be thrown before that